Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Sliders/PinchSlider.cs

```diff
@@ -0,0 +1,198 @@
+            get { return sliderValue; }
+            set
+            {
+                var oldSliderValue = sliderValue;
+                sliderValue = value;
+                UpdateUI();
+                OnValueUpdated.Invoke(new SliderEventData(oldSliderValue, value, activePointer is IMixedRealityPointer));
+            }
+        }
+
+        [Header("Slider Track")]
+
+        [Tooltip("The axis the slider moves along")]
+        [SerializeField]
+        private SliderAxis sliderAxis = SliderAxis.XAxis;
+        [Serializable]
+        /// <summary>
+        /// The axis the slider moves along
+        /// </summary>
+        private enum SliderAxis
+        {
+            XAxis = 0,
+            YAxis,
+            ZAxis
+        }
+
+        [SerializeField]
+        [Tooltip("Where the slider track starts, in local coordinates")]
+        private Vector3 sliderStartPosition = new Vector3(-0.5f, 0, 0);
+        [SerializeField]
+        [Tooltip("Where the slider track ends, in local coordinates")]
+        private Vector3 sliderEndPosition = new Vector3(0.5f, 0, 0);
+
+        /// <summary>
+        /// Gets the start position of the slider, in world space, or zero if invalid.
+        /// Sets the start position of the slider, in world space, projected to the slider's axis.
+        /// </summary>
+        public Vector3 SliderStartPosition
+        {
+            get { return transform.TransformPoint(sliderStartPosition); }
+            set { sliderStartPosition = Vector3.Project(transform.InverseTransformPoint(value), GetSliderAxis()); }
+        }
+
+        /// <summary>
+        /// Gets the end position of the slider, in world space, or zero if invalid.
+        /// Sets the end position of the slider, in world space, projected to the slider's axis.
+        /// </summary>
+        public Vector3 SliderEndPosition
+        {
+            get { return transform.TransformPoint(sliderEndPosition); }
+            set { sliderEndPosition = Vector3.Project(transform.InverseTransformPoint(value), GetSliderAxis()); }
+        }
+        #endregion
+
+
+        #region Event Handlers
+        [Header("Events")]
+        [Tooltip("Invoked when the component starts running")]
+        public SliderEvent OnStartRunning;
+        public SliderEvent OnValueUpdated;
+        public SliderEvent OnInteractionStarted;
+        public SliderEvent OnInteractionEnded;
+        public SliderEvent OnHoverEntered;
+        public SliderEvent OnHoverExited;
+        #endregion
+
+        #region Private Members
+        private float startSliderValue;
+        private Vector3 startPointerPosition;
+        private Vector3 startSliderPosition;
+        private Coroutine moveSliderCoroutine;
+        private IMixedRealityPointer activePointer;
+        #endregion
+
+
+        #region Unity methods
+        public void Start()
+        {
+            UpdateUI();
+            OnStartRunning.Invoke(new SliderEventData(sliderValue, sliderValue, false));
```

Comment:
What about raising a value updated event on start instead of having a start running one?

Reply:
Oh, great suggestion!